velox/exec/tests/TableWriteTest.cpp

```diff
@@ -93,6 +94,39 @@ class TableWriteTest : public HiveConnectorTestBase {
     return getRecursiveFiles(directoryPath).size();
   }
 
+  // Returns all available table types to test insert without any
+  // partitions (used in "immutablePartitions" set of tests).
+  std::vector<connector::hive::LocationHandle::TableType> tableTypes() {
+    return {
+        connector::hive::LocationHandle::TableType::kNew,
+        connector::hive::LocationHandle::TableType::kExisting};
+  }
+
+  // Returns a table insert plan node.
+  // Helper method used for "immutablePartitions" set of tests.
+  PlanNodePtr createInsertPlan(
+      const RowTypePtr& rowType,
+      const std::vector<RowVectorPtr>& vectors,
+      const connector::hive::LocationHandle::TableType& tableType,
+      const std::shared_ptr<TempDirectoryPath>& outputDirectory) {
```

Comment:
Why is this a TempDirectoryPath and not just a string for the directory ? Passing just "outputDirectory->path" here would make this cleaner I think.